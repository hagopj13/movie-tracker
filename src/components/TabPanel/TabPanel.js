// @flow
import React from 'react';
import type { Node } from 'react';

type Props = {
  value: number,
  index: number,
  children: Node,
};

const TabPanel = (props: Props) => {
  const { value, index, children } = props;

  return <div>{value === index && <>{children}</>}</div>;
};

export default TabPanel;
