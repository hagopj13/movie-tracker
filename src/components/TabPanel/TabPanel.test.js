import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import TabPanel from './TabPanel';

describe('TabPanel component', () => {
  let shallow;
  let value;
  let index;
  let wrapper;

  const ChildComponent = () => <div>Some child component</div>;

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    value = 0;
    index = 0;
    const mockProps = {
      value,
      index,
    };
    wrapper = shallow(
      <TabPanel {...mockProps}>
        <ChildComponent />
      </TabPanel>,
    );
  });

  it('should render the TabPanel component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the child component when value === index', () => {
    expect(wrapper.exists(ChildComponent)).toBe(true);
  });

  it('should not render the child comonent when value !== index', () => {
    const mockProps = {
      value: 0,
      index: 1,
    };
    const newWrapper = shallow(
      <TabPanel {...mockProps}>
        <ChildComponent />
      </TabPanel>,
    );
    expect(newWrapper.exists(ChildComponent)).toBe(false);
  });
});
