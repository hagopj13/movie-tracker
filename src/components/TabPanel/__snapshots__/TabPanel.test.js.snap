// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabPanel component should render the TabPanel component correctly 1`] = `
<div>
  <ChildComponent />
</div>
`;
